airflow/models/dag.py

```diff
@@ -112,6 +113,51 @@ def get_last_dagrun(dag_id, session, include_externally_triggered=False):
     return query.first()
 
 
+class _NextDagRunInfoLegacy(NamedTuple):
+    """Legacy return format for ``DAG.next_dagrun_info()``.
+
+    In the pre-AIP-39 implementation, ``DAG.next_dagrun_info()`` returns a
+    2-tuple ``(execution_date, run_after)``.
+    """
+
+    execution_date: Optional[pendulum.DateTime]
+    run_after: Optional[pendulum.DateTime]
+
+
+class _NextDagRunInfoCompat(_NextDagRunInfoLegacy):
```

Comment:
We don't need to maintain compat here (though doing so isn't all that hard) but I think we should give this a better/non-compat-or-private name.

I don't have any brilliant ideas for a name though.

Reply:
If it’s hard to describe, it’s probably a bad idea 🙂 I’m removing this altogether, we only need to fix a few tests anyway.